### ISSUE TITLE
Heatmap and metrics controls are not properly displayed on android

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-phone#78]
Both the heatmaps and metrics controls are not properly displayed. The cols for the year and month are really high, and so the overall width is too high and the right side is cut off. Compare the android and ios metrics screens below for an example. See how the android screen is missing the hour, the weekday and the summarization value.

| android (not working)  | ios (working) |
| ---------------------- | ---------------- |
| ![broken_controls](https://cloud.githubusercontent.com/assets/2423263/16903673/476a65e4-4c39-11e6-8fd9-6fccca65ad30.jpg)  | ![simulator screen shot jul 17 2016 9 47 51 am](https://cloud.githubusercontent.com/assets/2423263/16903632/7275eb38-4c38-11e6-9671-59d7b4c7692d.png)  |

**Comment by shankari** [comment]
The report from users like @gtfierro who have experienced this is that the controls used to work, at some point but they don't work any more. Gabe installed the app around Jun 9th, so version 0.0.9.

```
commit 9b0d445c1966bb2b3616f194e6de71ad1dc5f30a
Merge: f41eaed df544ac
Author: shankari <shankari@eecs.berkeley.edu>
Date:   Thu Jun 9 00:07:09 2016 -0700

    Merge pull request #37 from shankari/release_v_9
    
    Minor changes to kickstart the release cycle again

commit df544acf88bd40a9dbf8eb69c4f603a911378fce
Author: K. Shankari <shankari@eecs.berkeley.edu>
Date:   Wed Jun 8 15:20:44 2016 -0700

    Bump up version number to 0.0.9
```

**Comment by shankari** [comment]
The original thought was that this regression was due to 

```
commit 112a094980c915439b49299e6869b25cec2744f2
Author: K. Shankari <shankari@eecs.berkeley.edu>
Date:   Thu Jun 23 17:15:09 2016 -0700

    Create a non-crosswalk version for android 5.0+, which has a system updateable webview
    
    https://play.google.com/store/apps/details?id=com.google.android.webview
    https://infinum.co/the-capsized-eight/articles/the-updateable-webview-on-an
droid-5-lollipop-what-is-it-and-why-should-you-care
    
    Instructions from:
    https://github.com/crosswalk-project/cordova-plugin-crosswalk-webview#install
```

and the differences between the chromium webview and the xwalk webview, since all the examples were from people who had fairly recent versions of android.

However, I reverted back to the original webview, and the controls are still broken.

Now, my current top theory is the `@media` queries that @krisma added. Maybe the phones are so much bigger than the iPhones that they don't match any of the queries and that breaks the display for some reason.

**Comment by shankari** [comment]
After all this time, I am not actually sure that the heatmap controls ever worked. I have reverted back to commit 9b0d445c1966bb2b3616f194e6de71ad1dc5f30a, which should have been the version that worked - it was the version that corresponded to release 0.0.9.

If it really did work at that version, this is a regression. If not, then it never worked, and we just have to fix the layout on android.

**Comment by gtfierro** [comment]
![image](https://cloud.githubusercontent.com/assets/55434/16922887/5f3ad27e-4ccc-11e6-9c41-37d5efd599a8.png)

My heatmap and the new metrics tab both have the same UI bug that I noticed.

It definitely did work before, at least in the sense that there was a full date selector, and "Get!" returned some plotted data.

**Comment by shankari** [comment]
@gtfierro, "Get" should still return some data on both the heatmap and metrics screens - you just won't be able to select the day or weekday.

Also, I should clarify that I have not reverted back to commit 9b0d445 in the released apk, so it is expected behavior that the issue is still present in release 0.0.12.

I've only reverted back on a private branch that I have used to build an apk. @yw314cornell is running some power drain experiments on the Nexus 6, but once they are done today morning, she will be able to install the apk and verify that the controls did work then. After that, hopefully, we can start from a working version and do the binary search to figure out what broke and when.

**Comment by gtfierro** [comment]
Setting the "from" month to 1 and leaving the "to" month at 7 does not return data for either screen.

**Comment by shankari** [comment]
@gtfierro you can see from the screen photo at the beginning that "Get" does work, even with the broken controls :)

**Comment by shankari** [comment]
hm, that's unexpected. what happens if you kill the app, go to the metrics screen and then just hit "Get" without changing any of the controls.

I have to talk to the undergrads now, but will look on the server to see if I can see something from the logs right after that.

**Comment by gtfierro** [comment]
I have tried loading both the metrics and the heatmap after both killing the app from the task manager, Force Stop from the settings menu, and restarting the entire phone. Hitting "Get" in any of these scenarios does not plot any data for me, even if I adjust the "from" month. Could it be that the hidden day elements are set to -1 or 0 or some other value that creates an invalid date?

**Comment by shankari** [comment]
@gtfierro, try again.

it turned out that @yw314cornell was pulling a large amount of data using the public API and it crashed the webserver. That's what I get for trying to do something quick-and-dirty with bottle.

**Comment by gtfierro** [comment]
I can fetch data now, but "Stacked" on the metrics screen doesn't seem to change anything, and the y-axis says "Number" rather than some helpful units.

**Comment by shankari** [comment]
yeah other people have noticed that "Stacked" doesn't work. @krisma is going potentially change the visualization - I just pulled data and stuck into into a default bar graph with nvd3. The current thought is that it might not be bar graphs after all.

And the y axis is "Number" because the metric is "Count". Change the metric and the y-axis label will change too :)

**Comment by gtfierro** [comment]
But what is it a count of? Number of stops? Destinations? Trips? Kilosamples?

Its also not obvious that you can change the metric -- I didn't realize this until you implied it in your last message. Might be worth looking into some UI hint for that.

Thanks!

**Comment by shankari** [comment]
Good points! Will tell @krisma to take that into consideration when he does his design.

**Comment by shankari** [comment]
For the record, the controls screen is absolutely a regression. When David and @gtfierro first reported the issue, I remember showing that it worked on my screen. And now it doesn't. @krisma, I think that this is related to the `@media` stuff. May not be worth debugging if you are going to change the controls anyway.

**Comment by shankari** [comment]
Fixed by new filter screen implemented as part of #88
